### PR TITLE
[JW8-5650] Add support for Program-Date-Time metadata in Safari

### PR DESCRIPTION
**Pull request by robwalch**
### This PR will...
Add support for Program-Date-Time metadata in Safari

### Why is this Pull Request needed?
For Safari HLS playback with the html5 provider to have API parity with the HLS.js provider. Developers can use the PDT in these metadata events to calculate the running PDT using the start date and currentTime.

```js
{
    autostart: true,
    file: '//hls-stream-with-program-date-time.m3u8',
    events: {
        metadataCueParsed: function(e) {
            if (e.metadataType === 'program-date-time') {
                window.startDateTime = new Date(e.metadata.programDateTime).getTime() - e.metadata.start;
                console.log(e.type, 'program-date-time', e.metadata.programDateTime);
            }
        },
        meta: function(e) {
            if (e.metadataType === 'program-date-time') {
                console.log(e.type, 'program-date-time', e.metadata.programDateTime);
            }
        },
        time: function(e) {
            var fives = Math.floor(e.currentTime / 5);
            if (window.lastFive !== fives) {
                window.lastFive = fives;
                console.log(
                    'currentTime:', e.currentTime,
                    'program-date-time', new Date(e.currentTime + window.startDateTime).toISOString()
                );
            }
        }
    }
}
```

### Are there any Pull Requests open in other repos which need to be merged with this?
https://github.com/jwplayer/jwplayer-commercial/pull/6660

#### Addresses Issue(s):
JW8-5650

### CODE REVIEW

**Comment by johnBartos** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  🛠 There are modified src files, but no test changes. Add tests if you're able to.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/js/providers/html5.js [hunk 1]
```diff
@@ -294,9 +297,9 @@ function VideoProvider(_playerId, _playerConfig, mediaElement) {
     let dvrPosition = null;
     let dvrUpdatedTime = 0;
 
-    this.isSDK = !!_playerConfig.sdkplatform;
```
Comment (jnatalzia) [inline src/js/providers/html5.js]: is this not used anywhere?

File: src/js/providers/html5.js [hunk 2]
```diff
@@ -318,6 +321,32 @@ function VideoProvider(_playerId, _playerConfig, mediaElement) {
...
+                const { start, end } = _this.getSeekRange();
+                const metadataType = 'program-date-time';
+                const metadata = {
+                    metadataType,
+                    programDateTime,
+                    start,
+                    end
+                };
+                const cue = _this.createCue(start, end, JSON.stringify(metadata));
+                _this.addVTTCue({
```
Comment (jnatalzia) [inline src/js/providers/html5.js]: can i get a quick ELI5 on what cues do in this instance? i went and looked at the underlying function and my assumption is it's heavily reliant on the `metadata` flag but i'm not quite getting it. we don't show these in the ui right? is it a way of cueing up the information for the events we're sending in safari?

File: src/js/providers/html5.js (hunk 1 above)
Comment (robwalch) [reply src/js/providers/html5.js]: Nope. Spoke with @karimMourra and searched all projects for `isSDK`. html5.js is not used to handle playback, and if it was nothing in the player uses this property.

File: src/js/providers/html5.js (hunk 2 above)
Comment (robwalch) [reply src/js/providers/html5.js]: >  is it a way of cueing up the information for the events we're sending in safari?

Exactly. The video element has audio, video and text tracks. These text tracks can have a type of subtitles, chapters or metadata. These text tracks are made up of timed cues. When playback enters or exits a cue's range an event is fired notifying us that `activeCues` have changes. We use these events on the metadata track to fire "meta" events.

File: src/js/providers/html5.js [hunk 3]
```diff
@@ -318,6 +321,32 @@ function VideoProvider(_playerId, _playerConfig, mediaElement) {
         }
     }
 
+    function checkStartDateTime() {
+        if (_videotag.getStartDate) {
```
Comment (johnBartos) [inline src/js/providers/html5.js]: Nit: flip the conditional for an early return to reduce nesting

Comment (johnBartos) [reply src/js/providers/html5.js]: Its fine; build is failing since I merged the commercial PR before this.
